pkg/ds/daemon_set_test.go

```diff
@@ -289,28 +317,33 @@ func TestSchedule(t *testing.T) {
 
 	// 11 good nodes 11 bad nodes, and 1 good cherry picked node = 23 nodes
 	expectedNodes := 23
-	numNodes = waitForNodes(t, ds, expectedNodes, desiresErrCh, dsChangesErrCh)
-	Assert(t).AreEqual(numNodes, expectedNodes, "took too long to schedule")
+	err = waitForNodes(ds, expectedNodes, desiresErrCh, dsChangesErrCh)
+	Assert(t).IsNil(err, "took too long to schedule")
+
+	err = waitForPodsInIntent(consulStore, expectedNodes)
+	if err != nil {
+		t.Fatal(err)
+	}
 
 	//
-	// Deleting the daemon set should unschedule all of its nodes
+	// Deleting the daemon set should not unschedule any nodes
 	//
-	beforeDeletePods, _, err := scheduledPods(consulStore)
-	Assert(t).IsNil(err, "Unable to list intent/")
 	ds.logger.NoFields().Info("Deleting daemon set...")
 	err = dsStore.Delete(ds.ID())
 	if err != nil {
 		t.Fatalf("Unable to delete daemon set: %v", err)
 	}
 	// behavior change: Deleting a daemon set will no longer unschedule its pods (for now)
-	numNodes = waitForNodes(t, ds, 12, desiresErrCh, dsChangesErrCh)
-	Assert(t).AreEqual(numNodes, expectedNodes, "Unexpected number of nodes labeled")
+	err = waitForNodes(ds, 12, desiresErrCh, dsChangesErrCh)
```

Comment:
so now that you've made this error... how is it that it doesn't error? there really should be 23, right???